### PR TITLE
📝 [RUMF-1130] add migration guide to the CHANGELOG

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

Guide users who want to update from v3 to v4. [Rendered](https://github.com/DataDog/browser-sdk/blob/benoit/prepare-update-guide/CHANGELOG.md#v400)

## Changes

Add a migration guide in CHANGELOG

## Testing

- [ ] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,63 @@
...
+
+Here is the list of all the breaking changes introduced by v4 of the RUM and Logs SDKs.
+
+### [RUM and Logs] Intake URLs change
+
+We changed URLs of where the Browser SDK data is sent. Please make sure that your [Content Security Policy is up to date](https://docs.datadoghq.com/real_user_monitoring/faq/content_security_policy/?tab=us).
+
+### [RUM and Logs] Drop support for TypeScript below v3.8.2
+
+The Browser SDK is now incompatible with TypeScript v3.8.2. If you are using TypeScript, please make
```
Comment (bcaudan) [inline CHANGELOG.md:29]:
```suggestion
The Browser SDK is now incompatible with TypeScript below v3.8.2. If you are using TypeScript, please make
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,63 @@
...
+sure that the version you are using is greater or equal to v3.8.2.
+
+### [RUM and Logs] Tags syntax sanitization
+
+`version`, `env` and `service` [initialization parameters](https://docs.datadoghq.com/real_user_monitoring/browser/#initialization-parameters) are sent as tags to Datadog. The Browser SDK will now sanitize and print a warning if those values don't meet [tag requirements syntax](https://docs.datadoghq.com/getting_started/tagging/#defining-tags).
```
Comment (bcaudan) [inline CHANGELOG.md:34]:
```suggestion
`version`, `env` and `service` [initialization parameters](https://docs.datadoghq.com/real_user_monitoring/browser/#initialization-parameters) are sent as tags to Datadog. The Browser SDK is now sanitizing and printing a warning if those values don't meet [tag requirements syntax](https://docs.datadoghq.com/getting_started/tagging/#defining-tags).
```
afaik present tense is favored

File: CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,63 @@
...
+
+### [RUM and Logs] Configuration typings cleanup
+
+Some TypeScript types have been cleaned up to better represent accepted initialization parameters
+for each SDKs.
```
Comment (bcaudan) [inline CHANGELOG.md:39]: that could sound a bit cryptic from a customer point of view.
Do we expect them to change something?
if yes, what?
if no, does it worth to communicate on it?

Comment (BenoitZugmeyer) [reply CHANGELOG.md]: I clarified this section. I think it's worth mentioning it since it is still a breaking change. LMKWDYT

File: CHANGELOG.md [hunk 4]
```diff
@@ -16,6 +16,61 @@
...
+
+### [RUM and Logs] Drop support for TypeScript below v3.8.2
+
+The Browser SDK is now incompatible with TypeScript below v3.8.2. If you are using TypeScript, please make sure that the version you are using is greater or equal to v3.8.2.
+
+### [RUM and Logs] Tags syntax sanitization
+
+`version`, `env` and `service` [initialization parameters](https://docs.datadoghq.com/real_user_monitoring/browser/#initialization-parameters) are sent as tags to Datadog. The Browser SDK is now sanitizing and printing a warning if those values don't meet [tag requirements syntax](https://docs.datadoghq.com/getting_started/tagging/#defining-tags).
+
+### [RUM and Logs] Configuration typings cleanup
```
Comment (bcaudan) [inline CHANGELOG.md:35]:
```suggestion
### [RUM and Logs] Stricter initialization parameters typings
```

File: CHANGELOG.md (hunk 3 above)
Comment (bcaudan) [reply CHANGELOG.md]: Much clearer! I've added a follow up

**Review by bcaudan — APPROVED** [review]

**Review by amortemousque — APPROVED** [review]

**Pull request merged**